tests/container_checker/test_container_checker.py

```diff
@@ -0,0 +1,299 @@
+
+    pytest_assert(len(alerting_message["stdout_lines"]) > 0,
+                  "Failed to get Monit alerting message from container_checker!")
+
+    for container_name in stopped_containers_list:
+        if container_name not in alerting_message["stdout_lines"][0]:
+            pytest.fail("Container '{}' was not running and not found in Monit alerting message!"
+                        .format(container_name))
+
+
+def restart_containers(duthost, stopped_containers_list):
+    """Restarts the container after testing.
+
+    This function will restart the stopped containers and then check whether the
+    containers are actually restarted.
+
+    Args:
+        duthost: Host DUT.
+        stopped_containers_list: A list of stopped container names.
+
+    Return:
+        None.
+    """
+    for container_name in stopped_containers_list:
+        logger.info("Restarting the container '{}'...".format(container_name))
+        duthost.shell("sudo systemctl restart {}.service".format(container_name), module_ignore_errors=True)
+
+    for container_name in stopped_containers_list:
+        logger.info("Checking the running status of container '{}'...".format(container_name))
+        restarted = wait_until(CONTAINER_RESTART_THRESHOLD_SECS,
+                               CONTAINER_CHECK_INTERVAL_SECS,
+                               check_container_state, duthost, container_name, True)
+        if not restarted:
+            if is_hiting_start_limit(duthost, container_name):
+                clear_failed_flag_and_restart(duthost, container_name)
+            else:
+                pytest.fail("Failed to restart container '{}'".format(container_name))
+        logger.info("Container '{}' was restarted".format(container_name))
+
+
+def test_container_checker(duthosts, rand_one_dut_hostname, tbinfo):
+    """Tests the feature of container checker.
+
+    This function will check whether the container names will appear in the Monit
+    alerting message if they are stopped explicitly or they hit start limitation.
+
+    Args:
+        duthosts: list of DUTs.
+        rand_one_dut_hostname: hostname of DUT.
+        tbinfo: Testbed information.
+
+    Returns:
+        None.
+    """
+    duthost = duthosts[rand_one_dut_hostname]
+    container_autorestart_states = duthost.get_container_autorestart_states()
+    disabled_containers = get_disabled_container_list(duthost)
+
+    bgp_neighbors = duthost.get_bgp_neighbors()
+    up_bgp_neighbors = [ k.lower() for k, v in bgp_neighbors.items() if v["state"] == "established" ]
+
+    skip_containers = disabled_containers[:]
+    # Skip 'radv' container on devices whose role is not T0.
+    if tbinfo["topo"]["type"] != "t0":
+        skip_containers.append("radv")
+
+    stopped_containers_list = stop_containers(duthost, container_autorestart_states, skip_containers)
+    # Wait for 6 minutes such that Monit has a chance to write alerting message into syslog.
+    time.sleep(360)
```

Comment:
This is a long waiting time. Why this value?
Is there a way to probe expected message and fail eventually if timeout?

Reply:
Since the script `container_checker` was ran periodically by Monit and if one of containers was not running, Monit will write an alerting message into syslog after 5 minutes. So at here this pytest script has to wait for 6 minutes and then check whether the alerting message appeared in syslog or not.

Reply:
Sounds good.